pandas/core/arrays/string_.py

```diff
@@ -198,10 +198,9 @@ def _from_sequence(cls, scalars, dtype=None, copy=False):
         if dtype:
             assert dtype == "string"
 
-        result = np.asarray(scalars, dtype="object")
         # convert non-na-likes to str, and nan-likes to StringDtype.na_value
         result = lib.ensure_string_array(
-            result, na_value=StringDtype.na_value, copy=copy
+            scalars, na_value=StringDtype.na_value, copy=copy
```

Comment:
wow, do we need the code starting at L206 anymore then? (run asvs)

Reply:
Ran the asvs from benchmarks/strings.py a few times and there was either no change or a performance improvement in one run (maybe a false positive):
```
       before           after         ratio
     [d2e958b1]       [f082eef7]
     <master>         <float32-to-str-precision>
-      15.1±0.5ms       9.87±0.2ms     0.65  strings.Cat.time_cat(0, None, None, 0.0)

SOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.
PERFORMANCE INCREASED.
```
I think the lines below are needed because ensure_string_array returns an numpy array while we want a pandas array.